### PR TITLE
specs: Update `withdrawals` spec

### PR DESCRIPTION
**Pull request by clabby**
# Overview

Updates the `withdrawals` specification in accordance w/ #3836.

Note: The `L2OutputOracle`'s function names assume that #4045 will be merged.

### CODE REVIEW

File: specs/withdrawals.md [hunk 1]
```diff
@@ -55,18 +55,27 @@ We first describe the end to end flow of initiating and finalizing a withdrawal:
 ### On L2
 
 An L2 account sends a withdrawal message (and possibly also ETH) to the `L2ToL1MessagePasser` predeploy contract.
-   This is a very simple contract that stores the a hash of the withdrawal data.
+This is a very simple contract that stores the a hash of the withdrawal data.
 
 ### On L1
```
Comment (tynes) [inline specs/withdrawals.md:61]: Slight nit, markdown lists should always use `1` and the renderer will handle incrementing at runtime. This is to reduce diffs when making modifications

**Review by tynes — APPROVED** [review]

File: specs/withdrawals.md [hunk 2]
```diff
@@ -55,18 +55,27 @@ We first describe the end to end flow of initiating and finalizing a withdrawal:
...
+   These inputs include the withdrawal transaction data, inclusion proofs, and a block number. The block number
    must be one for which an L2 output root exists, which commits to the withdrawal as registered on L2.
-2. The `OptimismPortal` contract retrieves the output root for the given timestamp from the `OutputOracle`'s
-   `l2Outputs()` function, and performs the remainder of the verification process internally.
-3. If verification fails, the call reverts. Otherwise the call is forwarded, and the hash is recorded to prevent it from
-   from being replayed.
+1. The `OptimismPortal` contract retrieves the output root for the given block number from the `L2OutputOracle`'s
+   `getL2OutputAfter()` function, and performs the remainder of the verification process internally.
+1. If proof verification fails, the call reverts. Otherwise the call is forwarded, and the hash is recorded to
+   prevent it from being re-proven. Note that the withdrawal can be proven more than once if the corresponding
```
[the remark this answers is not in the document]
Comment (clabby) [reply specs/withdrawals.md:69]: Yes- The changes involved allowing for withdrawals to be re-proven only if the corresponding output root / output root proof changes. See: #4057

File: specs/withdrawals.md [hunk 3]
```diff
@@ -132,23 +141,22 @@ interface OptimismPortal {
 
     function l2Sender() returns(address) external;
 
+    function proveWithdrawalTransaction(
+        Types.WithdrawalTransaction memory _tx,
```
[the remark this answers is not in the document]
Comment (clabby) [reply specs/withdrawals.md:145]: Will add one in just a bit.

https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/libraries/Types.sol#L46-L56

**Review by mslipper — APPROVED** [review]

**Comment by maurelian** [comment]
@clabby I think you might need to rebase to get that devnet PR working

**Comment by mslipper** [comment]
@mergifyio refresh

File: specs/withdrawals.md [hunk 4]
```diff
@@ -55,18 +55,27 @@ We first describe the end to end flow of initiating and finalizing a withdrawal:
...
-   `l2Outputs()` function, and performs the remainder of the verification process internally.
-3. If verification fails, the call reverts. Otherwise the call is forwarded, and the hash is recorded to prevent it from
-   from being replayed.
+1. The `OptimismPortal` contract retrieves the output root for the given block number from the `L2OutputOracle`'s
+   `getL2OutputAfter()` function, and performs the remainder of the verification process internally.
+1. If proof verification fails, the call reverts. Otherwise the call is forwarded, and the hash is recorded to
+   prevent it from being re-proven. Note that the withdrawal can be proven more than once if the corresponding
+   output root changes.
+1. After the withdrawal is proven, it enters a 7 day challenge period, allowing time for other network participants
+   to challenge its integrity.
```
Comment (norswap) [inline specs/withdrawals.md:72]: @clabby Is it really possible to challenge a withdrawal? I think only output root can be challenged.

Comment (clabby) [reply specs/withdrawals.md:72]: @norswap This is what I meant, but I see how the wording is a bit unclear- Should we change this up?

Maybe:
```suggestion
1. After the withdrawal is proven, it enters a 7 day challenge period, allowing time for other network participants
   to challenge the integrity of the corresponding output root.
```

Comment (norswap) [reply specs/withdrawals.md:72]: Sounds great! Sorry for the w-e ping btw, just sorting through old bookmarks 😅

Comment (clabby) [reply specs/withdrawals.md:72]: No problem! Appreciate the nit, good spot 😄

**Pull request merged**